port/scorecard/resouce.go

```diff
@@ -0,0 +1,152 @@
+		return
+	}
+
+	err = refreshScorecardState(ctx, state, s, blueprintIdentifier)
+	if err != nil {
+		resp.Diagnostics.AddError("failed writing scorecard fields to resource", err.Error())
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *ScorecardResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state *ScorecardModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	s, err := scorecardResourceToPortBody(ctx, state)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to convert scorecard resource to body", err.Error())
+		return
+	}
+
+	sp, err := r.portClient.CreateScorecard(ctx, state.Blueprint.ValueString(), s)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create scorecard", err.Error())
+		return
+	}
+
+	writeScorecardComputedFieldsToState(state, sp)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func writeScorecardComputedFieldsToState(state *ScorecardModel, wp *cli.Scorecard) {
+	state.ID = types.StringValue(fmt.Sprintf("%s:%s", wp.Blueprint, wp.Identifier))
+	state.Identifier = types.StringValue(wp.Identifier)
+	state.CreatedAt = types.StringValue(wp.CreatedAt.String())
+	state.CreatedBy = types.StringValue(wp.CreatedBy)
+	state.UpdatedAt = types.StringValue(wp.UpdatedAt.String())
+	state.UpdatedBy = types.StringValue(wp.UpdatedBy)
+}
+
+func (r *ScorecardResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var state *ScorecardModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	s, err := scorecardResourceToPortBody(ctx, state)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to convert scorecard resource to body", err.Error())
+		return
+	}
+
+	sp, err := r.portClient.UpdateScorecard(ctx, state.Blueprint.ValueString(), state.Identifier.ValueString(), s)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to update the scorecard", err.Error())
+		return
+	}
+
+	writeScorecardComputedFieldsToState(state, sp)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *ScorecardResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state *ScorecardModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err := r.portClient.DeleteScorecard(ctx, state.Blueprint.ValueString(), state.Identifier.ValueString())
+
+	if err != nil {
+		resp.Diagnostics.AddError("failed to delete scorecard", err.Error())
+		return
+	}
+
+}
+
+func (r *ScorecardResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	idParts := strings.Split(req.ID, ":")
+
+	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
```

Comment:
consider making a regex for the format

Reply:
this is a standard https://github.com/hashicorp/terraform-provider-aws/blob/ffed9080eac29c568977990183faf0dd16a74ece/internal/service/wafv2/web_acl.go#L46